src/components/redirect.test.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import StaticRouter from 'react-router-dom/StaticRouter';
+import Redirect from './Redirect';
+
+const renderToContext = to => {
+	const context = {};
+
+	ReactDOMServer.renderToString(
+		<StaticRouter location={new URL('http://example.com')} context={context}>
+			<Redirect to={to} />
+		</StaticRouter>
+	);
+	return context;
+};
+
+describe('Server rendering', () => {
+	it('sets React Router context for "local" redirect, nothing from Redirect.rewind()', () => {
+		const internalTo = 'foo/bar';
+		const httpContext = renderToContext(internalTo);
+		expect(httpContext.url).toEqual(internalTo);
+		expect(Redirect.rewind()).toBeUndefined();
+	});
+	it('provides external redirect URL from Redirect.rewind(), nothing from React Router context', () => {
```

Comment:
You mean separate the https and http urls? I feel like that's not central to the behavior of the method, but I can split it out